pkg/util/cluster_variant.go

```diff
@@ -0,0 +1,24 @@
+package util
```

Comment:
It's not a good practice to overuse packages like `util`, `model` etc. in Go. Package names are part of the API. I'd prefer if we put this into a package name like `cluster`, and have `cluster.Variant` for enum and a method `cluster.DetectVariant()` to detect the variant.

See https://go.dev/blog/package-names